src/main/java/net/sf/marineapi/nmea/io/DefaultDataReader.java

```diff
@@ -52,6 +52,10 @@ public DefaultDataReader(InputStream source, SentenceReader parent) {
 	 */
 	@Override
 	public String read() throws Exception {
-		return input.readLine();
+		final String result = input.readLine();
+		if (result == null) {
+		    setRunning(false);

```

Comment:
From the BUfferedReader.readLine() javadoc:

     * @return     A String containing the contents of the line, not including
     *             any line-termination characters, or null if the end of the
     *             stream has been reached

And DefaultDataReader uses final BufferedReader input which is initialized as follows:

		this.input = new BufferedReader(isr);

With this, I cannot see how any more data could be received from the BufferedReader after having received null from readLine().